src/workerd/api/js-rpc-service.c++

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "js-rpc-service.h"
+
+namespace workerd::api {
+
+JsRpcServiceEvent::JsRpcServiceEvent()
+    : ExtendableEvent("jsRpcService") {};
+
+kj::Promise<WorkerInterface::CustomEvent::Result> JsRpcServiceCustomEventImpl::run(
+    kj::Own<IoContext::IncomingRequest> incomingRequest,
+    kj::Maybe<kj::StringPtr> entrypointName) {
+  // Set delivered, note that we're moving the IncomingRequest into the JsRpcTargetImpl (server),
+  // and we will drain once that object is destroyed. As of writing, this occurs after the first
+  // call() request completes (a "session" is a request/response).
+  incomingRequest->delivered();
+  server = kj::heap<JsRpcTargetImpl>(kj::mv(incomingRequest), kj::mv(entrypointName));
+
+  co_return WorkerInterface::CustomEvent::Result {
+    .outcome = EventOutcome::OK
+  };
```

Comment:
Given that this is no `co_await`ing anything, the `co_return` here (and below) is unnecessary.